NitroxServer/Communication/Packets/Processors/PlayerDeathEventProcessor.cs

```diff
@@ -0,0 +1,29 @@
+﻿using NitroxModel.Packets;
+using NitroxModel.Server;
+using NitroxServer.Communication.Packets.Processors.Abstract;
+using NitroxServer.GameLogic;
+
+namespace NitroxServer.Communication.Packets.Processors
+{
+    class PlayerDeathEventProcessor : AuthenticatedPacketProcessor<PlayerDeathEvent>
+    {
+        private readonly PlayerManager playerManager;
+        private readonly ServerConfig serverConfig;
+        public PlayerDeathEventProcessor(PlayerManager playerManager, ServerConfig serverConfig)
+        {
+            this.playerManager = playerManager;
+            this.serverConfig = serverConfig;
+        }
+
+        public override void Process(PlayerDeathEvent packet, Player player)
+        {
+            if(serverConfig.GameMode == "Hardcore")
+            {
+                player.IsPermaDeath = true;
+                PlayerKicked playerKicked = new PlayerKicked("Permanent death from hardcore mode");
+                player.SendPacket(playerKicked);
+            }
```

Comment:
Should we send a disconnect packet to other players?

Reply:
If we do that then it will kick all the other players though? 
My thoughts were that we only kick the player that has died permanently

Reply:
I wasn't suggesting to kick the other players.  I was saying to let players know that the player was kicked.

Though on second glance, this should happen anyways as PlayerKicked will send a disconnect automatically.